applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp

```diff
@@ -19,330 +20,401 @@
-    //***********************************************************************************
+    return 0;
+}
 
-    void BaseLoadCondition::AddExplicitContribution(const VectorType& rRHS,
-        const Variable<VectorType>& rRHSVariable,
-        Variable<array_1d<double,3> >& rDestinationVariable,
-        const ProcessInfo& rCurrentProcessInfo)
-    {
-        KRATOS_TRY
+/***********************************************************************************/
+/***********************************************************************************/
 
-        const unsigned int number_of_nodes = GetGeometry().PointsNumber();
-        const unsigned int dimension       = GetGeometry().WorkingSpaceDimension();
+double BaseLoadCondition::GetIntegrationWeight(
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints,
+    const SizeType PointNumber,
+    const double detJ
+    )
+{
+    return IntegrationPoints[PointNumber].Weight() * detJ;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseLoadCondition::AddExplicitContribution(
+    const VectorType& rRHS,
+    const Variable<VectorType>& rRHSVariable,
+    Variable<array_1d<double,3> >& rDestinationVariable,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
 
-        if( rRHSVariable == RESIDUAL_VECTOR && rDestinationVariable == FORCE_RESIDUAL )
-        {
+    const SizeType number_of_nodes = GetGeometry().PointsNumber();
+    const SizeType dimension       = GetGeometry().WorkingSpaceDimension();
 
-            for(SizeType i=0; i< number_of_nodes; i++)
-            {
-                SizeType index = dimension * i;
+    if( rRHSVariable == RESIDUAL_VECTOR && rDestinationVariable == FORCE_RESIDUAL ) {
+        for(SizeType i=0; i< number_of_nodes; ++i) {
+            SizeType index = dimension * i;
 
-                array_1d<double, 3 > &ForceResidual = GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
-                for(SizeType j=0; j<dimension; j++)
-                {
-                    #pragma omp atomic
-                    ForceResidual[j] += rRHS[index + j];
-                }
+            array_1d<double, 3 >& r_force_residual = GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
+            for(SizeType j=0; j<dimension; ++j) {
+                #pragma omp atomic
+                r_force_residual[j] += rRHS[index + j];
             }
         }
+    }
 
     KRATOS_CATCH( "" )
-    }
+}
+/***********************************************************************************/
+/***********************************************************************************/
 
-} // Namespace Kratos
+void BaseLoadCondition::save( Serializer& rSerializer ) const
+{
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, BaseType )
```

Comment:
I would have left this in the header ...
Dont know if it makes a difference, up to you